gem/views.py

```diff
@@ -304,3 +312,38 @@ def item_author_name(self, article_page):
 class GemAtomFeed(GemRssFeed):
     feed_type = Atom1Feed
     subtitle = GemRssFeed.description
+
+
+# https://github.com/praekelt/yal-merge/blob/develop/yal/views.py#L711-L751
+def clean_comment(self):
+    """
+    Check for email addresses and telephone numbers
+    """
+    comment = self.cleaned_data['comment']
+
+    site = Site.objects.get(is_default_site=True)
+    settings = GemSettings.for_site(site)
+
+    banned_list = [REGEX_EMAIL, REGEX_PHONE]
+
+    banned_keywords_and_patterns = \
+        settings.banned_keywords_and_patterns.split('\n') \
+        if settings.banned_keywords_and_patterns else []
+
+    banned_list += banned_keywords_and_patterns
+
+    for keyword in banned_list:
+        keyword = keyword.replace('\r', '')
+        match = re.search(keyword.lower(), comment.lower())
```

Comment:
@smn @miltontony How do you guys feel about keyword.lower()? I'm a bit concerned as this will change the meaning of regexes with special sequences (eg: \D). I'd be more comfortable not lowering the keyword and instructing users to write keywords as lowercase in the help text for the field on the GemSettings model.